src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -0,0 +1,230 @@
+
+(defn- scan-qr-code-tab
+  [qr-view-finder request-camera-permission camera-opts]
+  [:<>
+   [rn/view {:style style/scan-qr-code-container}]
+   (when (empty? @qr-view-finder)
+     [qr-scan-hole-area qr-view-finder])
+   (if (and @camera-permission-granted? (boolean (not-empty @qr-view-finder)))
+     [viewfinder @qr-view-finder camera-opts]
+     [camera-permission-view request-camera-permission])])
+
+(defn- enter-sync-code-tab
+  []
+  [rn/view {:style style/enter-sync-code-container}
+   [quo/text
+    {:size   :paragraph-1
+     :weight :medium
+     :style  {:color colors/white}}
+    "Yet to be implemented"]])
+
+(defn- bottom-view
+  [insets]
+  [rn/touchable-without-feedback
+   {:on-press #(js/alert "Yet to be implemented")}
+   [rn/view
+    {:style (style/bottom-container (:bottom insets))}
+    [quo/text
+     {:size   :paragraph-2
+      :weight :regular
+      :style  style/bottom-text}
+     (i18n/label :t/i-dont-have-status-on-another-device)]]])
+
+(defn- check-qr-code-data
+  [event]
+  (let [connection-string        (string/trim (oops/oget event "nativeEvent.codeStringValue"))
+        valid-connection-string? (sync-utils/valid-connection-string? connection-string)]
+    (if valid-connection-string?
+      (rf/dispatch [:syncing/input-connection-string-for-bootstrapping connection-string])
+      (rf/dispatch [:toasts/upsert
+                    {:icon           :i/info
+                     :icon-color     colors/danger-50
+                     :override-theme :light
+                     :text           (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
+
+(defn render-background
+  [background show-camera? qr-view-finder]
+  (if (and show-camera? (:x qr-view-finder))
+    [hole-view/hole-view
+     {:style style/absolute-fill
+      :holes [(merge qr-view-finder
+                     {:borderRadius 16})]}
+     background]
+    background)
+)
+
+(defn view
+  [{:keys [title show-bottom-view? background]}]
+  (let [active-tab                (reagent/atom 1)
+        qr-view-finder            (reagent/atom {})
+        request-camera-permission (fn []
+                                    (rf/dispatch
+                                     [:request-permissions
+                                      {:permissions [:camera]
+                                       :on-allowed  #(reset! camera-permission-granted? true)
+                                       :on-denied   #(rf/dispatch
+                                                      [:toasts/upsert
+                                                       {:icon :i/info
+                                                        :icon-color colors/danger-50
+                                                        :override-theme :light
+                                                        :text (i18n/label
+                                                               :t/camera-permission-denied)}])}]))]
+    [:f>
+     (fn []
```

Comment:
(fn [] should be never used in the hiccup. for proper usage, you have to rename `view` to `f-view` and used it with `[:f>`